### PR TITLE
Q6 adds getComments & fetchComments

### PR DESCRIPTION
**Pull request by rubyrubyruby02**

### CODE REVIEW

File: __tests__/integration.tests.js [hunk 1]
```diff
@@ -157,9 +157,49 @@ describe('Q5  adds /api/articles and returns articles in data-descending order',
...
         })
     })
 })
+describe('Q6 /api/articles/:article_id/comments', ()=> {
+    test('Status 200: responds with array with following properties: comment_id, votes, created_at, author, body, article_id', ()=> {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response)=> {
+            response.body.comments.forEach((comment)=> {
```
Comment (EddTheDuck) [inline __tests__/integration.tests.js:172]: If asserting inside a forEach you should also check the length of the array

File: __tests__/integration.tests.js [hunk 2]
```diff
@@ -157,9 +157,49 @@ describe('Q5  adds /api/articles and returns articles in data-descending order',
...
+        })
+    })
+    test('Status 200: to be sorted with most recent comment first by default', ()=> {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response)=> {
+            expect(response.body.comments).toBeSortedBy('created_at', {descending: true})
+        })
+    })
```
Comment (EddTheDuck) [inline __tests__/integration.tests.js:189]: What would happen when given a valid ID but there are no comments? ie. returns an empty array.

Are you satisfied that your code would handle this situation correctly?

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
Need to add an extra test but the rest looks great :)

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
 const {fetchComments} = require('../models/comments.model')
 
 const getComments = (request, response, next) => {
     const {article_id} = request.params
-    fetchComments(article_id).then((result)=> {
-        response.status(200).send({comments: result})
+
+    const promises = [fetchComments(article_id)]
+
+    if(article_id){
```
Comment (mezzDavies) [inline controllers/comments.controller.js:9]: Would there be a case when there wasn't an article_id? 🤔 

This controller has been invoked due to the matching url which has an article_id (valid or not) included. I'm pretty certain you can do away with this logic - refactor and see if it your test still passes. The beauty of tests! 👍

**Review by mezzDavies — APPROVED** [review]
Looks good you are welcome to merge. I have left one comment for your consideration - have a play around/refactor as I feel there is a bit of extraneous code 😄

**Pull request merged**